### PR TITLE
Fix and enhance CLI release workflow

### PR DESCRIPTION
**Pull request by riprasad**

### CODE REVIEW

File: .github/workflows/cli-release-linux.yaml [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
-        description: 'Release Version (e.g. 2.0.0.Final)'
-        required: true
+    types: [released]
 
 jobs:
   build:
+    if: github.repository_owner == 'Apicurio'
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
```
Comment (riprasad) [inline .github/workflows/cli-release-linux.yaml:16]: Existing workflow always checks out master which is not correct. The latest release was done from 2.0.x branch and therefore the build should have been done from the respective branch. But the CLI tools uploaded to the release asset was built from the master branch.

File: .github/workflows/cli-release-win64.yaml [hunk 2]
```diff
@@ -48,33 +52,52 @@ jobs:
...
-      shell: cmd
+
+    - name: Get maven wrapper
+      shell: bash
+      run: cd registry && mvn -N io.takari:maven:wrapper -Dmaven=3.6.3
+
     - name: Build jvm jar
-      run: mvn -pl cli -am install
+      shell: cmd
+      run: cd registry && mvnw.cmd -pl cli -am install
```
Comment (riprasad) [inline .github/workflows/cli-release-win64.yaml:72]: used maven wrapper

File: .github/workflows/cli-release-linux.yaml (hunk 1 above)
Comment (famartinrh) [reply .github/workflows/cli-release-linux.yaml:16]: when you trigger the workflow it asked for the branch to run the workflow on, I think it did checkout the code from the release branch

**Review by famartinrh — APPROVED** [review]
LGTM

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**